tests/Backend/Teradata/OrderByTest.php

```diff
@@ -0,0 +1,206 @@
+            ],
+            [
+                [
+                    'column' => 'column_string',
+                    'order' => 'DESC',
+                    'dataType' => 'non-existing',
+                ],
+                'Data type non-existing not recognized. Possible datatypes are [INTEGER|REAL]',
+            ],
+        ];
+    }
+
+    public function testNonArrayParamsShouldReturnErrorInAsyncExport(): void
+    {
+        // TODO enable when export supports filters
+        $this->markTestSkipped('export does not supports filters');
+//        $tableId = $this->prepareTable();
+//
+//        $orderBy = ['column' => 'column'];
+//
+//        $this->expectException(ClientException::class);
+//        $this->expectExceptionMessage("All items in param \"orderBy\" should be an arrays, but parameter contains:\n" . json_encode($orderBy));
+//        $this->getExportedTable($tableId, ['orderBy' => $orderBy]);
+    }
+
+
+    public function testInvalidStructuredQueryInAsyncExport(): void
+    {
+        // TODO enable when export supports filters
+        $this->markTestSkipped('export does not supports filters');
+//        $tableId = $this->prepareTable();
+//
+//        $orderBy = 'string';
+//
+//        $this->expectException(ClientException::class);
+//        $this->expectExceptionMessage("Parameter \"orderBy\" should be an array, but parameter contains:\n" . json_encode($orderBy));
+//        $this->getExportedTable($tableId, ['orderBy' => $orderBy]);
+    }
+
+    public function testNonArrayParamsShouldReturnErrorInDataPreview(): void
+    {
+        $tableId = $this->prepareTable();
+
+        $orderBy = ['column' => 'column'];
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("All items in param \"orderBy\" should be an arrays, but parameter contains:\n" . json_encode($orderBy));
+        $this->_client->getTableDataPreview($tableId, ['orderBy' => $orderBy]);
+    }
+
+
+    public function testInvalidStructuredQueryInADataPreview(): void
+    {
+        $tableId = $this->prepareTable();
+
+        $orderBy = 'string';
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage("Parameter \"orderBy\" should be an array, but parameter contains:\n" . json_encode($orderBy));
+        $this->_client->getTableDataPreview($tableId, ['orderBy' => $orderBy]);
+    }
+
+
+    private function getExportedTable(string $tableId, array $exportOptions): array
+    {
+        $tableExporter = new TableExporter($this->_client);
+        $path = tempnam(sys_get_temp_dir(), 'keboola-export');
+        $this->assertIsString($path);
+        $tableExporter->exportTable($tableId, $path, $exportOptions);
+        $fileContents = file_get_contents($path);
+        $this->assertIsString($fileContents);
+        return Client::parseCsv($fileContents);
+    }
+
+    private function prepareTable(): string
```

Comment:
nechces na to pouzit nejake existujici CSV nebo ho pridat do testovacich dat? Tento in-time CSV building mi neprijde dobry